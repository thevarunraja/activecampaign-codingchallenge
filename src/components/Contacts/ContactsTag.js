import React from "react";
import axios from "axios";

export default function ContactsTag({ contactTagsLink }) {
  const [tags, setTags] = React.useState("");

  React.useEffect(() => {
    async function fetchTags() {
      const res = await axios.get(
        `${contactTagsLink.split("https://lamppoststudios.api-us1.com/")[1]}`
      );
      if (res.data.contactTags.length >= 1) {
        var tags = res.data.contactTags.map(tag => tag.tag);
        setTags(tags.join(", "));
      }
    }
    fetchTags();
  }, [contactTagsLink]);
  return <div>{tags}</div>;
}
