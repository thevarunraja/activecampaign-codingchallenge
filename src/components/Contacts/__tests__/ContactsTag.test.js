import "@testing-library/jest-dom/extend-expect";
import { cleanup, render, wait } from "@testing-library/react";
import axiosMock from "axios";
import React from "react";
import ContactsTag from "../ContactsTag";

afterEach(cleanup);

describe("ContactsTag Component", () => {
  it("fetched and displays data", async () => {
    axiosMock.get.mockResolvedValueOnce({
      data: {
        contactTags: [
          {
            contact: "856",
            tag: "7",
            cdate: "2019-02-21T10:22:02-06:00",
            created_timestamp: "2019-08-14 15:54:19",
            updated_timestamp: "2019-08-14 15:54:19",
            created_by: null,
            updated_by: null,
            links: {
              tag:
                "https://lamppoststudios.api-us1.com/api/3/contactTags/839/tag",
              contact:
                "https://lamppoststudios.api-us1.com/api/3/contactTags/839/contact"
            },
            id: "839"
          }
        ]
      }
    });
    const { container } = render(
      <ContactsTag
        contactTagsLink={
          "https://lamppoststudios.api-us1.com/api/3/contacts/856/contactTags"
        }
      />
    );
    await wait(() => {
      expect(axiosMock.get).toHaveBeenCalledTimes(1);
      expect(container).toMatchSnapshot();
    });
  });
});
