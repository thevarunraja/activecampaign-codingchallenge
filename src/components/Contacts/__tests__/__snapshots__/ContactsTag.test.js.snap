// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsTag Component fetched and displays data 1`] = `
<div>
  <div>
    7
  </div>
</div>
`;
